modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/AMQPConstructStore.java

```diff
@@ -122,15 +122,26 @@ public void addQueue(AndesQueue queue, boolean isLocal) throws AndesException {
      * remove a queue
      *
      * @param queueName name of the queue to be removed
-     * @param isLocal   is this a local change
      * @throws AndesException
      */
-    public void removeQueue(String queueName, boolean isLocal) throws AndesException {
-        if (isLocal) {
-            andesContextStore.deleteQueueInformation(queueName);
-            //create the space created to keep message counter on this queue
-            messageStore.removeQueue(queueName);
-        }
+    public void removeLocalQueue(String queueName) throws AndesException {
+
+        // Remove the queue from internal maps
+        removeClusterQueue(queueName);
```

Comment:
Similarly after invalidating the cache before removing the entry some can read the cache. When that happens the old queue mapping is reloaded and will be used even after the db entry is deleted. Therefore it seems changing the order won't solve the issue.

Alternative approaches I can think of are,

1. Use a read write lock to make the operations atomic
2. Use a hash instead of an auto incremented id for queue mapping.